<launch>

  <arg name="marker_id"			default="26"/>
  <arg name="marker_size"		default="0.070"/>    <!-- in meters -->
  <arg name="camera_frame"		default=""/>
  <arg name="marker_frame"		default="marker_frame"/>
  <arg name="reference_frame"		default=""/>
  <arg name="corner_refinement"		default="LINES"/>
  <arg name="planarity_tolerance"	default="0.001"/>
  <arg name="manager"			default=""/>
  <arg name="camera_info_topic"		default="camera_info"/>
  <arg name="image_topic"		default="texture"/>
  <arg name="depth_topic"		default="depth_map"/>

  <node if="$(eval manager == '')"
	name="aruco_detector" pkg="aist_aruco_ros" type="aist_aruco_detector"
  	output="screen">
    <remap from="/camera_info"		to="$(arg camera_info_topic)"/>
    <remap from="/image"		to="$(arg image_topic)"/>
    <remap from="/depth"		to="$(arg depth_topic)"/>

    <!-- Other parameters -->
    <param name="image_is_rectified"	value="false"/>
    <param name="marker_id"		value="$(arg marker_id)"/>
    <param name="marker_size"		value="$(arg marker_size)"/>
    <param name="camera_frame"		value="$(arg camera_frame)"/>
    <param name="marker_frame"		value="$(arg marker_frame)"/>
    <param name="reference_frame"	value="$(arg reference_frame)"/>
    <param name="corner_refinement"	value="$(arg corner_refinement)"/>
    <param name="planarity_tolerance"	value="$(arg planarity_tolerance)"/>
  </node>

  <node unless="$(eval manager == '')"
	name="aruco_detector" pkg="nodelet" type="nodelet" output="screen"
  	args="load aist_aruco_ros/aist_aruco_detector_nodelet $(arg manager)">
    <remap from="/camera_info"		to="$(arg camera_info_topic)"/>
    <remap from="/image"		to="$(arg image_topic)"/>
    <remap from="/depth"		to="$(arg depth_topic)"/>

    <!-- Other parameters -->
    <param name="image_is_rectified"	value="false"/>
    <param name="marker_id"		value="$(arg marker_id)"/>
    <param name="marker_size"		value="$(arg marker_size)"/>
    <param name="camera_frame"		value="$(arg camera_frame)"/>
    <param name="marker_frame"		value="$(arg marker_frame)"/>
    <param name="reference_frame"	value="$(arg reference_frame)"/>
    <param name="corner_refinement"	value="$(arg corner_refinement)"/>
    <param name="planarity_tolerance"	value="$(arg planarity_tolerance)"/>
  </node>

</launch>
